src/web-animations-next-animation.js

```diff
@@ -173,14 +180,14 @@
       this._removeChildren();
     },
     reverse: function() {
+      var oldCurrentTime = this.currentTime;
       this._animation.reverse();
-      scope.awaitStartTime(this);
-      this._register();
-      this._forEachChild(function(child, offset) {
-        child.reverse();
-        child.startTime = this.startTime + offset * this.playbackRate;
-        child.currentTime = this.currentTime + offset * this.playbackRate;
+      this._forEachChild(function(childAnimation) {
+        childAnimation.reverse();
```

Comment:
> I don't understand why this works :p Is it because we set the currentTime on 189?

I don't know why you think it shouldn't work :P But yeah, the offset of the children is re-applied in the currentTime setter.